q2_feature_classifier/_skl.py

```diff
@@ -6,11 +6,49 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+from dataclasses import dataclass, field
+from functools import cached_property
 from itertools import islice, repeat
-from copy import deepcopy
+from typing import Dict, List
 
 from joblib import Parallel, delayed
 
+
+@dataclass
+class _TaxonPredictionNode:
+    name: str
+    offset: int # The offset index of this taxon in the probability vector
+    children: Dict[str, "_TaxonPredictionNode"] = field(default_factory=dict, repr=False)
+
+    @classmethod
+    def create_tree(cls, classes: List[str]):
+        root = cls("Unassigned", 0)
+        for i, label in enumerate(classes):
+            taxons = label.split(';')
```

Comment:
Please pass the `separator` that is passed to `predict` down to this level, so that it is guaranteed to be consistent.